internal/buffer/unbounded.go

```diff
@@ -47,6 +48,10 @@ func NewUnbounded() *Unbounded {
 // Put adds t to the unbounded buffer.
 func (b *Unbounded) Put(t interface{}) {
 	b.mu.Lock()
+	if b.closed {
+		b.mu.Unlock()
```

Comment:
Is it possible to defer the `Unlock()` instead?

Reply:
It is. I just don't like abusing `defer` in small, simple functions. And it saves a few nanoseconds. In that case it's not really important though.
If you find using `defer` here more readable we can use it instead.

Reply:
It used to be the case that `defer`'s performance was an issue back in the day. The Go team has done a lot of work on improving this, and we generally prefer using `defer` for unlocks whenever there is more than one way to exit the function.